exercises/atbash-cipher/example.ml

```diff
@@ -0,0 +1,57 @@
+let substitute = function
+  | 'a' .. 'z' as c ->
+    let offset = (Char.code c) - (Char.code 'a') in
+    let code = (Char.code 'z') - offset in
+    Char.chr code
+  | other -> other
+
+let is_encodable = function
+  | 'a' .. 'z' | '0' .. '9' -> true
+  | _ -> false
+
+let explode s =
+  let rec expl i l =
+    if i < 0 then
+      l
+    else
+      expl (i - 1) (s.[i] :: l)
+  in
+  expl ((String.length s) - 1) []
+
+let implode l =
+  let character_at i = List.nth l i in
+  String.init (List.length l) character_at
+
+let rec take n l =
+  match (n, l) with
+  | (0, _) -> []
+  | (_, []) -> []
+  | (_, x::xs) -> x::(take (n-1) xs)
+
+let rec drop n l =
+  match (n, l) with
+  | (0, _) -> l
+  | (_, []) -> []
+  | (_, _::xs) -> drop (n-1) xs
+
+let rec group n l =
+  if (List.length l) < n then
+    [l]
+  else
+    (take n l) :: (group n (drop n l))
+
+let encode ?block_size:(block_size = 5) word =
```

Comment:
Perhaps ```text``` is better than ```word``` - the input can be more than 1 word.

Also, the optional parameter can be provided more simply:
```?(block_size = 5)```